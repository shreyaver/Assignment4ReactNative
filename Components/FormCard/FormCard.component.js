import React, {Component} from 'react';
import {
  Text, View,
} from 'react-native';
import PropTypes from 'prop-types';
import styles from './FormCard.component.stylesheet';
import convertAndFormatDateToLocalTime from '../../helpers/convertAndFormatDateToLocalTime/convertAndFormatDateToLocalTime';

export default class FormCard extends Component {
  render() {
    const { form } = this.props;
    return (
        <View style={styles.card}>
          <View style={styles.formNameContainer}>
            <Text style={styles.formName} numberOfLines={3}>
              {form.name}
            </Text>
          </View>
          <View style={styles.formCreatedContainer}>
            <Text style={styles.createdAt}>
              Created on:
              {' '}
              {convertAndFormatDateToLocalTime(form.createdAt)}
            </Text>
          </View>
        </View>);
  } 
}
FormCard.propTypes = {
  form: PropTypes.shape({
    name: PropTypes.string.isRequired,
    createdAt: PropTypes.instanceOf(Date).isRequired,
  }).isRequired,
};