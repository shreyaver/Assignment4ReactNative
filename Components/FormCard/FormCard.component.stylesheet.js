import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  card: {
    width: '75%',
    backgroundColor: 'white',
    marginBottom: 25,
  },
  formNameContainer: {
    paddingVertical: 20,
    paddingHorizontal: 10,
  },
  formName: {
    fontSize: 22,
    color: 'black',
    height: 80,
  },
  formCreatedContainer: {
    alignItems: 'flex-end',
    borderTopWidth: 1,
    padding: 10,
  },
  createdAt: {
    fontSize: 15,
    color: 'black',
  },
});
