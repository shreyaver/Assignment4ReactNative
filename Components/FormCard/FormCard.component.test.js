import React from 'react';
import renderer from 'react-test-renderer';
import FormCard from './FormCard.component';

describe('FormCard', () => {
  const mockFormCardData = {
      name: 'Personal Details',
      createdAt: new Date('2019-01-01T12:00:00'),
  };
  it('renders without crashing', () => {
    const formCard = renderer.create(<FormCard form={mockFormCardData}/>).toJSON();
    expect(formCard).toMatchSnapshot();
  });
});