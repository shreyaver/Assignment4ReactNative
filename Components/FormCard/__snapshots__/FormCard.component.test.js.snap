// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormCard renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "marginBottom": 25,
      "width": "75%",
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 10,
        "paddingVertical": 20,
      }
    }
  >
    <Text
      numberOfLines={3}
      style={
        Object {
          "color": "black",
          "fontSize": 22,
          "height": 80,
        }
      }
    >
      Personal Details
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "borderTopWidth": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 15,
        }
      }
    >
      Created on:
       
      01/01/2019 12:00:00
    </Text>
  </View>
</View>
`;
